### PR TITLE
[#172838730] Updates profileFirstLogin condition to dispatch and payload to mixpanel

### PR DESCRIPTION
**Pull request by CrisTofani**
**Short description:**
This PR removes payload of profileFirstLogin action and changes the condition to dispatch

**List of changes proposed in this pull request:**
- ts/sagas/startup/checkProfileEnabledSaga.ts
- ts/store/actions/profile.ts
- ts/store/middlewares/analytics.ts

### CODE REVIEW

**Comment by Undermaken** [comment]
@CrisTofani did you test it?
Can you see this event in Mixpanel?

**Comment by CrisTofani** [comment]
> @CrisTofani did you test it?
> Can you see this event in Mixpanel?

@Undermaken i didn't, sorry i forgot to set the status to draft, I still didn't set as finished on Pivotal as a personal memo but should have done it here too. 

Anyway I now have finished testing, the actions are visible on pivotal too without any payload

File: ts/sagas/startup/checkProfileEnabledSaga.ts [hunk 1]
```diff
@@ -10,17 +10,11 @@ export function* checkProfileEnabledSaga(
   profile: InitializedProfile
 ): IterableIterator<Effect> {
   if (
-    !profile.has_profile ||
+    profile.version === 0 ||
```
Comment (Undermaken) [inline ts/sagas/startup/checkProfileEnabledSaga.ts:13]: I guess we can split this in 2 blocks.
A block checks for **is_inbox_enabled** and **is_webhook_enabled** (as it is we could call an upsert when it is not needed)
Another block check if `profile.version === 0` and if yes dispatch the action

Also the tests inside **_checkProfileEnabledSaga.test.ts_** should be updated

Comment (CrisTofani) [reply ts/sagas/startup/checkProfileEnabledSaga.ts:13]: Yes, I was checking the CircleCI script right now

File: ts/sagas/startup/checkProfileEnabledSaga.ts [hunk 2]
```diff
@@ -10,17 +10,9 @@ export function* checkProfileEnabledSaga(
   profile: InitializedProfile
 ): IterableIterator<Effect> {
   if (
-    !profile.has_profile ||
-    !profile.email ||
-    !profile.is_inbox_enabled ||
-    !profile.is_webhook_enabled
+    profile.version === 0 &&
```
Comment (Undermaken) [inline ts/sagas/startup/checkProfileEnabledSaga.ts:13]: Here we can use `isProfileFirstOnBoarding` selector
Check if we use this raw check somewhere and replace it ;)

Comment (CrisTofani) [reply ts/sagas/startup/checkProfileEnabledSaga.ts:13]: I noticed that it is a utility check not a real selector, I find out also this other utility check: `hasProfileEmail` I replaced the `!profile.email` with `!hasProfileEmail(profile)` too.

Comment (CrisTofani) [reply ts/sagas/startup/checkProfileEnabledSaga.ts]: the replacements are in 4d8e093

**Pull request merged**